internal/buildscripts/packaging/fpm/tar/build.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+
+# Copyright Splunk Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -euxo pipefail
+
+SCRIPT_DIR="$( cd "$( dirname ${BASH_SOURCE[0]} )" && pwd )"
+. $SCRIPT_DIR/../common.sh
+
+VERSION="${1:-}"
+ARCH="${2:-amd64}"
+OUTPUT_DIR="${3:-$REPO_DIR/dist}"
+SMART_AGENT_RELEASE="${4:-}"
+BUNDLE_BASE_DIR="/splunk-otel-collector"
+AGENT_BUNDLE_INSTALL_DIR="$BUNDLE_BASE_DIR/agent-bundle"
+OTELCOL_INSTALL_PATH="$BUNDLE_BASE_DIR/bin/otelcol"
+TRANSLATESFX_INSTALL_PATH="$BUNDLE_BASE_DIR/bin/translatesfx"
+
+tar_download_smart_agent() {
+    local tag="$1"
+    local buildroot="$2"
+    local api_url=""
+    local dl_url=""
+
+    if [ "$tag" = "latest" ]; then
+        tag=$( curl -sL "$SMART_AGENT_RELEASE_URL/latest" | jq -r '.tag_name' )
+        if [ -z "$tag" ]; then
+            echo "Failed to get tag_name for latest release from $SMART_AGENT_RELEASE_URL/latest" >&2
+            exit 1
+        fi
+    fi
+
+    api_url="$SMART_AGENT_RELEASE_URL/tags/$tag"
+    dl_url="$( curl -sL "$api_url" | jq -r '.assets[] .browser_download_url' | grep "signalfx-agent-${tag#v}\.tar\.gz" )"
+    if [ -z "$dl_url" ]; then
+        echo "Failed to get the agent download url from $api_url" >&2
+        exit 1
+    fi
+
+    echo "Downloading $dl_url ..."
+    curl -sL "$dl_url" -o "$buildroot/signalfx-agent.tar.gz"
+
+    mkdir -p "$buildroot/$BUNDLE_BASE_DIR"
+    tar -xzf "$buildroot/signalfx-agent.tar.gz" -C "$buildroot/"
+    mv "$buildroot/signalfx-agent" "$buildroot/$AGENT_BUNDLE_INSTALL_DIR"
+    find "$buildroot/$AGENT_BUNDLE_INSTALL_DIR" -wholename "*test*.key" -delete -or -wholename "*test*.pem" -delete
```

Comment:
Should we also exclude the `signalfx-agent` binary and `agent-status` symlink, similar to https://github.com/signalfx/splunk-otel-collector/blob/main/internal/buildscripts/packaging/fpm/common.sh#L98-L99 with the deb/rpm packages?

Reply:
Good catch!